File: module_me/src/main/java/com/ebook/me/MyCommentActivity.java
package com.ebook.me;


import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModelProvider;

import com.ebook.common.event.KeyCode;
import com.ebook.common.view.DeleteDialog;
import com.ebook.me.adapter.CommentListAdapter;
import com.ebook.me.databinding.ActivityCommentBinding;
import com.ebook.me.mvvm.factory.MeViewModelFactory;
import com.ebook.me.mvvm.viewmodel.CommentViewModel;
import com.scwang.smart.refresh.layout.api.RefreshLayout;
import com.therouter.TheRouter;
import com.therouter.router.Route;
import com.xrn1997.common.mvvm.view.BaseMvvmRefreshActivity;
import com.xrn1997.common.util.ObservableListUtil;

@Route(path = KeyCode.Me.COMMENT_PATH, params = {"needLogin", "true"})
public class MyCommentActivity extends BaseMvvmRefreshActivity<ActivityCommentBinding, CommentViewModel> {

    @Override
    public int onBindLayout() {
        return R.layout.activity_comment;
    }

    @NonNull
    @Override
    public Class<CommentViewModel> onBindViewModel() {
        return CommentViewModel.class;
    }

    @NonNull
    @Override
    public ViewModelProvider.Factory onBindViewModelFactory() {
        return MeViewModelFactory.INSTANCE;
    }

    @Override
    public void initViewObservable() {

    }

    @Override
    public int onBindVariableId() {
        return BR.viewModel;
    }

    @Override
    public void initView() {
        CommentListAdapter mCommentListAdapter = new CommentListAdapter(this, mViewModel.mList);
        mViewModel.mList.addOnListChangedCallback(ObservableListUtil.getListChangedCallback(mCommentListAdapter));
        getBinding().viewMyCommentList.setAdapter(mCommentListAdapter);
        mCommentListAdapter.setOnItemClickListener((comment, position) -> {
            Bundle bundle = new Bundle();
            bundle.putString("chapterUrl", comment.getChapterUrl());
            bundle.putString("chapterName", comment.getChapterName());
            bundle.putString("bookName", comment.getBookName());
            TheRouter.build(KeyCode.Book.COMMENT_PATH)
                    .with(bundle)
                    .navigation(MyCommentActivity.this);
        });
        mCommentListAdapter.setOnItemLongClickListener((comment, position) -> {
            DeleteDialog deleteDialog = DeleteDialog.newInstance();
            deleteDialog.setOnClickListener(() -> mViewModel.deleteComment(comment.getId()));
            deleteDialog.show(getSupportFragmentManager(), "deleteDialog");
            return true;
        });
    }

    @Override
    public boolean enableToolbar() {
        return true;
    }

    @Override
    public void initData() {
        mViewModel.refreshData();
    }

    @NonNull
    @Override
    public RefreshLayout getRefreshLayout() {
        return getBinding().refreshCommentList;
    }
}

File: module_me/src/main/java/com/ebook/me/MyCommentActivity.kt
package com.ebook.me

import android.os.Bundle
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.activity.viewModels
import androidx.recyclerview.widget.LinearLayoutManager
import com.ebook.api.entity.Comment
import com.ebook.common.event.KeyCode
import com.ebook.common.view.DeleteDialog.Companion.newInstance
import com.ebook.me.adapter.CommentListAdapter
import com.ebook.me.databinding.ActivityCommentBinding
import com.ebook.me.mvvm.viewmodel.CommentViewModel
import com.scwang.smart.refresh.layout.api.RefreshLayout
import com.therouter.TheRouter.build
import com.therouter.router.Route
import com.xrn1997.common.mvvm.view.BaseMvvmRefreshActivity
import dagger.hilt.android.AndroidEntryPoint

@AndroidEntryPoint
@Route(path = KeyCode.Me.COMMENT_PATH, params = ["needLogin", "true"])
class MyCommentActivity : BaseMvvmRefreshActivity<ActivityCommentBinding, CommentViewModel>() {
    override val mViewModel: CommentViewModel by viewModels()
    override fun initView() {
        val mCommentListAdapter = CommentListAdapter(this)
        mViewModel.mList.observe(this) {
            mCommentListAdapter.submitList(it)
        }
        binding.viewMyCommentList.layoutManager = LinearLayoutManager(this)
        binding.viewMyCommentList.adapter = mCommentListAdapter
        mCommentListAdapter.setOnItemClickListener { comment: Comment, _: Int ->
            val bundle = Bundle()
            bundle.putString("chapterUrl", comment.chapterUrl)
            bundle.putString("chapterName", comment.chapterName)
            bundle.putString("bookName", comment.bookName)
            build(KeyCode.Book.COMMENT_PATH)
                .with(bundle)
                .navigation(this@MyCommentActivity)
        }
        mCommentListAdapter.setOnItemLongClickListener { comment: Comment, _: Int ->
            val deleteDialog = newInstance()
            deleteDialog.setOnClickListener { mViewModel.deleteComment(comment.id) }
            deleteDialog.show(supportFragmentManager, "deleteDialog")
            true
        }
    }

    override fun enableToolbar(): Boolean {
        return true
    }
    override fun enableLoadMore(): Boolean {
        return false
    }

    override fun initData() {
        mRefreshLayout.autoRefresh()
    }

    override fun onBindViewBinding(
        inflater: LayoutInflater,
        parent: ViewGroup?,
        attachToParent: Boolean
    ): ActivityCommentBinding {
        return ActivityCommentBinding.inflate(inflater, parent, attachToParent)
    }

    override fun getRefreshLayout(): RefreshLayout {
        return binding.refreshCommentList
    }
}
